src/customConfigTests/src/test/kotlin/org/vaccineimpact/orderlyweb/customConfigTests/CustomConfigTests.kt

```diff
@@ -48,6 +50,29 @@ abstract class CustomConfigTests
         }
     }
 
+    private fun changeConfig(defaultBranchOnly: Boolean)
+    {
+        val mapper = ObjectMapper(YAMLFactory())
+        val configFile = File("${AppConfig()["orderly.root"]}/orderly_config.yml")
+        val oldConfig = mapper.readValue(configFile, OrderlyConfig::class.java)
+
+        if (oldConfig.remote!!["main"]?.default_branch_only != defaultBranchOnly)
+        {
+            val newRemote = oldConfig.remote!!["main"]!!.copy(default_branch_only = defaultBranchOnly)
+            val newConfig = oldConfig.copy(remote = linkedMapOf("main" to newRemote))
+            configFile.writeText(mapper.writeValueAsString(newConfig))
+        }
+        HttpClient.post("http://localhost:8321/v1/reload/")
+    }
+
+    @BeforeEach
+    fun beforeEach(info: TestInfo)
+    {
+        createDatabase()
+        val defaultBranchOnly = info.displayName.contains("DEFAULT_BRANCH_ONLY")
```

Comment:
This seems surprising - the only way to get the enum value for the parameterized test here is via `info.displayName`? I suppose they're not expecting the parameterized value to be needed outside the test body itself.

Reply:
I know, we ran into this with HINT and it's ugly/suprising. There will be some way to get this working using reflection but complicated! See this SO https://stackoverflow.com/questions/62036724/how-to-parameterize-beforeeach-in-junit-5

Maybe worth doing though if it makes usage much easier...

Reply:
Not sure if worth it if this is working ok for us, but maybe merits a code comment. I guess the main thing is to keep it localised to one place in the base class as you have done so that if we do decide we need to change it later it's only needed in one place!